core/ssh_keys_test.go

```diff
@@ -0,0 +1,453 @@
+	tc := testclient.New(nil, nil, nil)
+	c := NewSSHKeysClient(tc)
+	assert.Equal(t, tc, c.client)
+	assert.Equal(t, &url.URL{Path: "/core/v1/"}, c.basePath)
+}
+
+func TestSSHKeysClient_List(t *testing.T) {
+	type args struct {
+		ctx  context.Context
+		org  OrganizationRef
+		opts *ListOptions
+	}
+	tests := []struct {
+		name     string
+		args     args
+		resp     *katapult.Response
+		respErr  error
+		respV    *sshKeysResponseBody
+		want     []*AuthSSHKey
+		wantReq  *katapult.Request
+		wantResp *katapult.Response
+		wantErr  string
+	}{
+		{
+			name: "success",
+			args: args{
+				ctx: context.Background(),
+				org: OrganizationRef{ID: "org_O648YDMEYeLmqdmn"},
+				opts: &ListOptions{
+					Page:    5,
+					PerPage: 32,
+				},
+			},
+			resp: &katapult.Response{
+				Response: &http.Response{StatusCode: http.StatusOK},
+			},
+			respV: &sshKeysResponseBody{
+				Pagination: &katapult.Pagination{Total: 394},
+				SSHKeys: []*AuthSSHKey{
+					{ID: "ssh_O574YEEEYeLmqdmn"},
+				},
+			},
+			want: []*AuthSSHKey{
+				{ID: "ssh_O574YEEEYeLmqdmn"},
+			},
+			wantReq: &katapult.Request{
+				Method: "GET",
+				URL: &url.URL{
+					Path: "/core/v1/organizations/_/ssh_keys",
+					RawQuery: url.Values{
+						"page":     []string{"5"},
+						"per_page": []string{"32"},
+						"organization[id]": []string{
+							"org_O648YDMEYeLmqdmn",
+						},
+					}.Encode(),
+				},
+			},
+			wantResp: &katapult.Response{
+				Pagination: &katapult.Pagination{Total: 394},
+				Response:   &http.Response{StatusCode: http.StatusOK},
+			},
+		},
+		{
+			name: "success with nil options",
+			args: args{
+				ctx: context.Background(),
+				org: OrganizationRef{ID: "org_O648YDMEYeLmqdmn"},
+			},
+			resp: &katapult.Response{
+				Pagination: &katapult.Pagination{Total: 333},
+			},
+			respV: &sshKeysResponseBody{
+				SSHKeys: []*AuthSSHKey{
+					{ID: "ssh_O574YEEEYeLmqdmn"},
+				},
+			},
+			want: []*AuthSSHKey{
+				{ID: "ssh_O574YEEEYeLmqdmn"},
+			},
+			wantReq: &katapult.Request{
+				Method: "GET",
+				URL: &url.URL{
+					Path: "/core/v1/organizations/_/ssh_keys",
+					RawQuery: url.Values{
+						"organization[id]": []string{
+							"org_O648YDMEYeLmqdmn",
+						},
+					}.Encode(),
+				},
+			},
+		},
+		{
+			name: "request error",
+			args: args{
+				ctx: context.Background(),
+				org: OrganizationRef{ID: "org_O648YDMEYeLmqdmn"},
+			},
+			respErr: fmt.Errorf("flux capacitor undercharged"),
+			wantErr: "flux capacitor undercharged",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tc := testclient.New(tt.resp, tt.respErr, tt.respV)
+			c := NewSSHKeysClient(tc)
+			ctx := test.Context(tt.args.ctx)
+
+			got, resp, err := c.List(ctx, tt.args.org, tt.args.opts)
+
+			assert.Equal(t, 1, len(tc.Calls), "only 1 request should be made")
+			test.AssertContext(t, ctx, tc.Ctx)
+
+			assert.Equal(t, tt.want, got)
+
+			if tt.resp != nil {
+				assert.Equal(t, tt.resp, resp)
+			}
```

Comment:
This assertion needs updating to make use the `wantResp` tt value:

```suggestion
			if tt.wantResp != nil {
				assert.Equal(t, tt.wantResp, resp)
			}
```